pkg/chunk/azure/blob_storage_client.go

```diff
@@ -17,11 +17,25 @@ import (
 	"github.com/cortexproject/cortex/pkg/util/flagext"
 )
 
+//blob service endpoint of AzureGlobal
 const blobURLFmt = "https://%s.blob.core.windows.net/%s/%s"
 const containerURLFmt = "https://%s.blob.core.windows.net/%s"
 
+// blob service endpoint of AzureChinaCloud
+const chinaBlobURLFmt = "https://%s.blob.core.chinacloudapi.cn/%s/%s"
+const chinaContainerURLFmt = "https://%s.blob.core.chinacloudapi.cn/%s"
+
+// blob service endpoint of AzureGermanCloud
+const germanyBlobURLFmt = "https://%s.blob.core.cloudapi.de/%s/%s"
+const germanyContainerURLFmt = "https://%s.blob.core.cloudapi.de/%s"
+
+// blob service endpoint of AzureUSGovernment
+const govBlobURLFmt = "https://%s.blob.core.usgovcloudapi.net/%s/%s"
+const govContainerURLFmt = "https://%s.blob.core.usgovcloudapi.net/%s"
+
 // BlobStorageConfig defines the configurable flags that can be defined when using azure blob storage.
 type BlobStorageConfig struct {
```

Comment:
Could you add a `func (cfg * BlobStorageConfig) Validate() error` function to check if the environment value is one of the supported ones, please? Then this `Validate()` function should be called by the `Validate()` functions on the parent configs (structs using this config). I've checked the (2) parent configs and they already support the `Validate()`, so should be an easy thing to do.

Reply:
done